### PR TITLE
Add time plotting python script

### PR DESCRIPTION
**Pull request by Piphi5**
This adds a time plotting python script that plots data over time. This is mainly intended to help troubleshoot a scenario in which a dataset breaks sysid. It supports both json and csv plotting.

Example plots:
General Mech:
image

Drivetrain:
image

### CODE REVIEW

File: scripts/time_plots.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        columns.extend([
+            "Volts (V)", f"Position ({abbreviation})",
+            f"Velocity ({abbreviation}/s)"
+        ])
+        unit_columns = columns[3:]
+
+    df = pd.DataFrame(columns=columns)
+    for test in (test for test in data.keys() if "-" in test):
+        df_data = [[pt[0]] + [test] + pt[1:] for pt in data[test]]
+        df = df.append(pd.DataFrame(df_data, columns=columns))
```
Comment (calcmogul) [inline scripts/time_plots.py:53]: Isn't assigning to `df` redundant here?
```
>>> df=["a"]
>>> df.append("b")
>>> df
['a', 'b']
```

File: scripts/time_plots.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        ])
+        unit_columns = columns[4:8]
+    else:
+        columns.extend([
+            "Volts (V)", f"Position ({abbreviation})",
+            f"Velocity ({abbreviation}/s)"
+        ])
+        unit_columns = columns[3:]
+
+    df = pd.DataFrame(columns=columns)
```
Comment (calcmogul) [inline scripts/time_plots.py:50]: `df` could use a more descriptive name.

File: scripts/time_plots.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import pandas as pd
```
Comment (calcmogul) [inline scripts/time_plots.py:1]: Sorta these alphabetically if possible.

**Review by calcmogul — CHANGES_REQUESTED** [review]

File: scripts/time_plots.py (hunk 1 above)
Comment (Piphi5) [reply scripts/time_plots.py:53]: by default the append method returns a copy, but I can set the in_place param to True if that would reduce the redundancy

**Comment by prateekma** [comment]
@calcmogul Can you review again?

File: scripts/time_plots.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
+import json
```
Comment (calcmogul) [inline scripts/time_plots.py:1]: Put `#!/usr/bin/env python3` on the first line of the file, then a blank line. Also, mark time_plots.py as executable by running `git update-index --chmod=+x time_plots.py`.

File: scripts/time_plots.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+    # Get Unit
+    if unit == "Meters":
+        abbreviation = "m"
+    elif unit == "Feet":
+        abbreviation = "ft"
+    elif unit == "Inches":
+        abbreviation = "in"
+    elif unit == "Degrees":
+        abbreviation = "deg"
```
Comment (calcmogul) [inline scripts/time_plots.py:23]: "Radians" should be added as a valid unit with the abbreviation "rad".

**Review by calcmogul — CHANGES_REQUESTED** [review]

**Comment by calcmogul** [comment]
Are you sure you committed the change to the execute bit?

**Review by calcmogul — APPROVED** [review]

**Comment by Piphi5** [comment]
Should be updated now

**Pull request merged**